Symfony/CS/Fixer/Symfony/MethodSeparationFixer.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Symfony;
+
+use SplFileInfo;
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+use Symfony\CS\Tokenizer\TokensAnalyzer;
+
+/**
+ * Make sure there is one blank line above and below a method.
+ */
+final class MethodSeparationFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Methods in a class, interface or trait must be separated with one blank line.';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // Must run before braces and indentation fixers because this fixer
+        // might add line breaks to the code without indenting.
+        return 55;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(Token::getClassyTokenKinds());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(SplFileInfo $file, Tokens $tokens)
+    {
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+
+        for ($index = $tokens->getSize() - 1; $index > 0;--$index) {
+            if (!$tokens[$index]->isClassy()) {
+                continue;
+            }
+
+            // figure out where the classy starts
+            $classStart = $tokens->getNextTokenOfKind($index, array('{'));
+
+            // figure out where the classy ends
+            $classEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $classStart);
+
+            if ($tokens[$index]->isGivenKind(T_CLASS) || (defined('T_TRAIT') && $tokens[$index]->isGivenKind(T_TRAIT))) {
+                $this->fixClass($tokens, $tokensAnalyzer, $classStart, $classEnd);
+            } elseif ($tokens[$index]->isGivenKind(T_INTERFACE)) {
+                $this->fixInterface($tokens, $classStart, $classEnd);
+            }
```

Comment:
No it does not hurt. I would prefer the simplification, but you can leave it as is. ;)

(When a new classy type is added we have to check all `isClassy` usages anyway.)